### ISSUE TITLE
JsonDecodingException: Polymorphic serializer was not found for missing class discriminator

### ISSUE THREAD

**Issue by PetrBe** [rjaros/kvision#185]
Hello, thanks for this great project, it's really cool!
A problem I'm facing. I use  kotlin serialization polymorphism in my project. And if try to get object with polymorphic hierarchy, from backend, I get this error.

example:

1. kvservice:

```kotlin
@KVService
interface IUserService {
    suspend fun getItemGroups(deviceId: Int): List<ItemsGroup>
}
```
2. implementation of this interface:
```kotlin
override suspend fun getItemGroups(deviceId: Int): List<ItemsGroup> {
        return call.withUser { user ->
            dbQuery {
                DeviceRepo.getDeviceItemGroupsWithStat()
            }
        }
    }
```

3. ItemsGroup object:
```kotlin
@Serializable
data class ItemsGroup(
        val id: Int? = null,
        var items: List<Item>? = null,
        val name: String? = ""
)
```

4. and ItemsGroup contains object with serializable hierarchy - **Item**:
```kotlin
@Serializable
sealed class Item {
    var id: Int? = null
    var name: String? = null
}
```

5. finally I`m trying call getItemGroups(1) that return ItemsGroup with **Item** subclass:
```kotlin
@Serializable
class ItemSubclass(val testField:String) : Item()
```

sampe of this ItemsGroup:
```kotlin
data class ItemsGroup(
        val id: Int? = null,
        var items: List<ItemSubclass>? = null,
        val name: String? = ""
)
```

and get error
JsonDecodingException: Polymorphic serializer was not found for missing class discriminator

If I serialize/deserialize `List<ItemsGroup>` object manually, using kotlin serialization - it`s work fine.

**Comment by rjaros** [comment]
Hello and thx for your feedback :-)

This issue was in fact already reported as #139 and was not resolved then.
But now I've tried a different way. Using your example code I've tried to make Jackson better cooperate with kotlinx.serialization. And I think I've managed to fix this. With the new KVision release it will be fully possible to use sealed classes both as parameters and return values, with additional `@JsonTypeInfo` annotation for the class, like this:
```
@JsonTypeInfo(Id.CLASS, As.PROPERTY, "type", JsonTypeInfo::class, false)
@Serializable
sealed class Item {
    var id: Int? = null
    var name: String? = null
}
```
